composer/callbacks/system_metrics_monitor.py

```diff
@@ -46,9 +86,25 @@ def run_event(self, event: Event, state: State, logger: Logger):
         ]:
             local_node_system_metrics = self.compute_system_metrics()
             all_system_metrics = dist.all_gather_object(local_node_system_metrics)
-            system_metrics = {
-                key: value for local_metrics in all_system_metrics for key, value in local_metrics.items()
-            }
+            system_metrics = {}
+            gpu_metrics_set = set(_GPU_METRICS)
+
+            if self.log_all_data:
+                for rank, metrics in enumerate(all_system_metrics):
+                    for key, value in metrics.items():
+                        if key in gpu_metrics_set:
+                            system_metrics[f'Rank_{rank}_{key}'] = value
```

Comment:
```suggestion
                            system_metrics[f'{key}_rank_{rank}'] = value
```
it's probably easier to view when sorted by key vs. rank and this lets us standardize formatting across the two log calls